packages/playwright-test/src/reporters/azure.ts

```diff
@@ -0,0 +1,481 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type * as Test from 'azure-devops-node-api/TestApi';
+import type * as TestInterfaces from 'azure-devops-node-api/interfaces/TestInterfaces';
+import * as azdev from 'azure-devops-node-api';
+
+import type { Reporter, TestCase, TestResult } from '@playwright/test/reporter';
+
+import type { TestPoint } from 'azure-devops-node-api/interfaces/TestInterfaces';
+import type { WebApi } from 'azure-devops-node-api';
+import { colors } from 'playwright-core/lib/utilsBundle';
+import { existsSync, readFileSync } from 'fs';
+import type { ICoreApi } from 'azure-devops-node-api/CoreApi';
+import type { TeamProject } from 'azure-devops-node-api/interfaces/CoreInterfaces';
+import { createGuid } from 'playwright-core/lib/utils';
+
+const Statuses = {
+  passed: 'Passed',
+  failed: 'Failed',
+  skipped: 'Paused',
+  timedOut: 'Failed'
+};
+
+const attachmentTypesArray = [
+  'screenshot',
+  'video',
+  'trace',
+] as const;
+
+type TAttachmentType = Array<typeof attachmentTypesArray[number]>;
+
+export interface AzureReporterOptions {
+  token: string;
+  planId: number;
+  orgUrl: string;
+  projectName: string;
+  logging?: boolean | undefined;
+  isDisabled?: boolean | undefined;
+  environment?: string | undefined;
+  testRunTitle?: string | undefined;
+  uploadAttachments?: boolean | undefined;
+  attachmentsType?: TAttachmentType | undefined;
+}
+
+interface TestResultsToTestRun {
+  statusCode: number;
+  result: Result;
+  headers: Headers;
+}
+interface Result {
+  count: number;
+  value?: ValueEntity[] | null;
+}
+interface ValueEntity {
+  id: number;
+  project: Project;
+  outcome: string;
+  testRun: TestRun;
+  priority: number;
+  url: string;
+  lastUpdatedBy: LastUpdatedBy;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface Project {
+}
+interface TestRun {
+  id: string;
+}
+interface LastUpdatedBy {
+  displayName?: null;
+  id?: null;
+}
+interface Headers {
+  'cache-control': string;
+  'pragma': string;
+  'content-length': string;
+  'content-type': string;
+  'expires': string;
+  'p3p': string;
+  'x-tfs-processid': string;
+  'strict-transport-security': string;
+  'activityid': string;
+  'x-tfs-session': string;
+  'x-vss-e2eid': string;
+  'x-vss-senderdeploymentid': string;
+  'x-vss-userdata': string;
+  'x-frame-options': string;
+  'request-context': string;
+  'access-control-expose-headers': string;
+  'x-content-type-options': string;
+  'x-cache': string;
+  'x-msedge-ref': string;
+  'date': string;
+  'connection': string;
+}
+
+class AzureDevOpsReporter implements Reporter {
+  private testApi!: Test.ITestApi;
+  private coreApi!: ICoreApi;
+  private _options: AzureReporterOptions;
```

Comment:
you don't need this since you already "preprocess" the options and store them as properties inside the class.

Reply:
done